src/main/java/com/cleanroommc/groovyscript/compat/mods/prodigytech/ExplosionFurnaceAdditives.java

```diff
@@ -0,0 +1,130 @@
+import com.cleanroommc.groovyscript.api.documentation.annotations.MethodDescription;
+import com.cleanroommc.groovyscript.api.documentation.annotations.RegistryDescription;
+import com.cleanroommc.groovyscript.helper.ingredient.ItemsIngredient;
+import com.cleanroommc.groovyscript.helper.ingredient.OreDictIngredient;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import lykrast.prodigytech.common.recipe.ExplosionFurnaceManager;
+import net.minecraft.item.ItemStack;
+
+@RegistryDescription(category = RegistryDescription.Category.ENTRIES)
+public class ExplosionFurnaceAdditives extends VirtualizedRegistry<ExplosionFurnaceAdditives.EFAdditiveRecipe> {
+    @Override
+    public void onReload() {
+        removeScripted().forEach(EFAdditiveRecipe::unregister);
+        restoreFromBackup().forEach(EFAdditiveRecipe::register);
+    }
+
+    private void add(EFAdditiveRecipe recipe) {
+        recipe.register();
+        addScripted(recipe);
+    }
+
+    private void remove(EFAdditiveRecipe recipe) {
+        recipe.unregister();
+        addBackup(recipe);
+    }
+
+    @MethodDescription(example = @Example("item('minecraft:cobblestone'), 50"), type = MethodDescription.Type.ADDITION)
+    public void addExplosive(IIngredient explosive, int power) {
+        EFAdditiveRecipe recipe = new EFAdditiveRecipe(true, explosive, power);
+        add(recipe);
+    }
+
+    @MethodDescription(example = @Example("ore('gunpowder')"))
+    public boolean removeExplosive(IIngredient explosive) {
+        for (ItemStack it : explosive.getMatchingStacks()) {
+            ExplosionFurnaceManager.Explosive externalExplosive = ExplosionFurnaceManager.findExplosive(it);
+            EFAdditiveRecipe recipe = new EFAdditiveRecipe(true, explosive, externalExplosive.getPower());
+            remove(recipe);
+            return true;
+        }
+        return false;
+    }
+
+    @MethodDescription(example = @Example(priority = 2000, commented = true))
+    public void removeAllExplosives() {
+        ExplosionFurnaceManager.EXPLOSIVES.getAllContent().forEach(r ->
+            addBackup(new EFAdditiveRecipe(true, new ItemsIngredient(r.getMatchingStacks()), r.getPower())));
+        ExplosionFurnaceManager.removeAllExplosives();
+    }
+
+    @MethodDescription(example = @Example("item('minecraft:stone'), 50"), type = MethodDescription.Type.ADDITION)
+    public void addDampener(IIngredient dampener, int power) {
+        EFAdditiveRecipe recipe = new EFAdditiveRecipe(false, dampener, power);
+        add(recipe);
+    }
+
+    @MethodDescription(example = @Example("ore('dustAsh')"))
+    public boolean removeDampener(IIngredient dampener) {
+        for (ItemStack it : dampener.getMatchingStacks()) {
+            ExplosionFurnaceManager.Dampener externalDampener = ExplosionFurnaceManager.findDampener(it);
+            EFAdditiveRecipe recipe = new EFAdditiveRecipe(false, dampener, externalDampener.getDampening());

```

Comment:
if `ExplosionFurnaceManager.findDampener(it)` doesnt find anything, it returns `null`, which then generates a `NullPointerException` on this line. this also happens to `ExplosionFurnaceManager.findExplosive(it)`.
there should be an `if (externalDampener == null) continue;` to avoid this.